exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofMapContainsMatcher.java

```diff
@@ -59,7 +59,11 @@ protected void describeMismatchSafely(ProofMapIndexProxy<HashCode, String> map,
   private CheckedMapProof checkProof(ProofMapIndexProxy<HashCode, String> map) {
     UncheckedMapProof proof = map.getProof(key);
     assert proof != null : "The proof must not be null";
-    return proof.check();
+
+    CheckedMapProof checkedProof = proof.check();
+    assert checkedProof.compareWithRootHash(map.getRootHash())
```

Comment:
Please use the matcher for that to report differences properly and not depend on enabled assertions — that is something that must always be checked. It is not an assertion for both MapIndex & Proof are SuTs.